core/src/main/java/com/opensymphony/xwork2/util/fs/DefaultFileManager.java

```diff
@@ -40,14 +40,30 @@
     private static final Pattern JAR_PATTERN = Pattern.compile("^(jar:|wsjar:|zip:|vfsfile:|code-source:)?(file:)?(.*?)(\\!/|\\.jar/)(.*)");
     private static final int JAR_FILE_PATH = 3;
 
-    protected static Map<String, Revision> files = Collections.synchronizedMap(new HashMap<String, Revision>());
+    protected static final Map<String, Revision> files = Collections.synchronizedMap(new HashMap<String, Revision>());
 
     protected boolean reloadingConfigs = false;
 
     public DefaultFileManager() {
     }
 
     public void setReloadingConfigs(boolean reloadingConfigs) {
+        if (reloadingConfigs && !this.reloadingConfigs) {
+            this.reloadingConfigs = true;
+            //starting monitoring not-monitored loaded files on demand
+            synchronized (files) {
+                for (String fileName :
+                        files.keySet()) {
+                    if (null == files.get(fileName)) {
```

Comment:
Hm... right now when I saw this I think it's really doesn't make sense to put `null`s in the `files` - you will get exactly the same behaviour when the list will be empty. So this basically means (in the `monitorFile` below)

```java
        if (!reloadingConfigs) {
            return;
        }
```

instead of

```java
        if (!reloadingConfigs) {
            files.put(fileName, null);
            return;
        }
```